pyomo/repn/parameterized_linear.py

```diff
@@ -0,0 +1,400 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2024
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import copy
+import enum
+
+from pyomo.common.collections import ComponentSet
+from pyomo.common.enums import ExtendedEnumType
+from pyomo.common.numeric_types import native_numeric_types
+from pyomo.core import Var
+from pyomo.core.expr.logical_expr import _flattened
+from pyomo.core.expr.numeric_expr import (
+    AbsExpression,
+    DivisionExpression,
+    LinearExpression,
+    MonomialTermExpression,
+    NegationExpression,
+    mutable_expression,
+    PowExpression,
+    ProductExpression,
+    SumExpression,
+    UnaryFunctionExpression,
+)
+from pyomo.repn.linear import (
+    ExitNodeDispatcher,
+    _initialize_exit_node_dispatcher,
+    LinearBeforeChildDispatcher,
+    LinearRepn,
+    LinearRepnVisitor,
+)
+from pyomo.repn.util import ExprType
+from . import linear
```

Comment:
This is a strange mix of relative and absolute imports.  Maybe `import pyomo.repn.linear as linear`?